### ISSUE TITLE
Incorrect evaluation in some cases when there midnight rollover

### ISSUE THREAD

**Issue by vpatel-gb** [opening-hours/opening_hours.js#261]
Steps to reproduce:

- https://openingh.ypid.de/evaluation_tool/ (or writing own code also gives same results)
- Specify this opening_hours string: `Mo 06:00-15:00; Tu-Th 07:00-04:00; Fr-Su 09:00-17:00`
- As shown in the screenshot, the `Tue` has incorrect interpretation. It should be `open from 7am to 4am`. Instead, it shows `7am to 12am` (i.e. 12am to 4am is missed)
- Worth noting that, exactly same scenario for `Wed` and `Thu` works fine.

![image](https://user-images.githubusercontent.com/22038192/40740059-09b81b84-6405-11e8-9c68-998e34b64f54.png)

**Comment by HolgerJeromin** [comment]
Isn't the marked place exactly right? the "rest" of the "Tu" hours are attached to the "We" night.
On the other hand i miss 0:00 - 04:00 on friday, but this is probably the warning it shows.

**Comment by vpatel-gb** [comment]
Sorry, there was some error in explaining the issue. 
`Updated the description` @HolgerJeromin

**Comment by vpatel-gb** [comment]
Thanks for the explanation about `Friday` missing info. So while converting our UI to `open_hour` format, we need to specifically add logic so that `Mo 06:00-15:00; Tu-Th 07:00-04:00; Fr-Su 09:00-17:00` should be generated as `Mo 06:00-15:00; Fr-Su 09:00-17:00; Tu-Th 07:00-04:00;`.

Is there any other way to tell the library that don't override when there is midnight-rollover?

**Comment by ypid** [comment]
> Is there any other way to tell the library that don't override when there is midnight-rollover?

No, there is not. This is the expected behavior which has often been discussed. You will need to use `,` as the tool kindly tells you in the warning :)
